src/components/radio/Radio.tsx

```diff
@@ -0,0 +1,46 @@
+import { Component, createSignal, Show } from 'solid-js'
+
+import StyledIcon from './StyledIcon'
+import StyledInput from './StyledInput'
+import StyledLabel from './StyledRadio'
+
+export type IRadioProps = {
+  label?: any
+  labelPosition?: 'right' | 'left'
+  checked?: boolean
+  indeterminate?: boolean
+  disabled?: boolean
+  className?: string
+  onChange?: (checked: boolean) => void
+}
+
+const Radio: Component<IRadioProps> = (props) => {
```

Comment:
While testing, I did:
```ts
const [isChecked, setIsChecked] = createSignal(false)

setTimeout(() => {
  setIsChecked(true)
}, 2000)

return <Radio checked={isChecked()} />
```
This should set the checked to true after 2 seconds, but it doesn't respect the checked prop correctly.

You might just need to add a:
```ts
createEffect(() => {
  if (props.checked != null) {
    setIsChecked(props.checked)
  }
})
```

Reply:
I think we need to add `createComputed` as `createEffect` will run on every change